### PR TITLE
Migrate Fullscreen mode test to Playwright

### PR DESCRIPTION
**Pull request by alvitazwar**
## What?
Based on https://github.com/WordPress/gutenberg/pull/38570, part of https://github.com/WordPress/gutenberg/issues/38851. Migrate [fullscreen-mode.test.js](https://github.com/alvitazwar/gutenberg/blob/add/playwright-fullscreen/test/e2e/specs/editor/various/fullscreen-mode.spec.js) to its Playwright counterpart.
## Why?
See https://github.com/WordPress/gutenberg/pull/38570 for its background and rationale.

This is split into a new PR for easier review.

## How?
See https://github.com/WordPress/gutenberg/pull/38570 for the proposed migration steps.

## Testing Instructions
npm run test:e2e:playwright /test/e2e/specs/editor/various/fullscreen-mode.spec.js

## Screencast
https://user-images.githubusercontent.com/55917380/193243524-249e9dc1-d1fc-40c7-b3e8-8bec2e89d4e1.mov

### CODE REVIEW

**Comment by alvitazwar** [comment]
@ntwb, @nerrad  can you please review this?

**Comment by alvitazwar** [comment]
@kevin940726  Can you please review this PR?

File: test/e2e/specs/editor/various/fullscreen-mode.spec.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Full Screen', () => {
```
Comment (Mamaduka) [inline test/e2e/specs/editor/various/fullscreen-mode.spec.js]:
```suggestion
test.describe( 'Fullscreen Mode', () => {
```

Suggestion: Let's use the same name as the original test.

File: test/e2e/specs/editor/various/fullscreen-mode.spec.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllPosts();
+	} );
+
+	test( 'should open the fullscreen mode from the more menu', async ( {
+		page,
+	} ) => {
+		// Open Options
+		await page.locator( ' [aria-label="Options"]' ).click();
```
Comment (Mamaduka) [inline test/e2e/specs/editor/various/fullscreen-mode.spec.js]:
```suggestion
		await page.click(
			'role=region[name="Editor top bar"i] >> role=button[name="Options"i]'
		);
```

Suggestion: The "Options" can be a common name for the button, so let's be more specific by using the region.

File: test/e2e/specs/editor/various/fullscreen-mode.spec.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+		// Verify All options are visible
+		page.locator( '.interface-more-menu-dropdown__content' );
```
Comment (Mamaduka) [inline test/e2e/specs/editor/various/fullscreen-mode.spec.js]:
```suggestion
```

Suggestion: The locators don't perform any assertions themself. Feel free to remove this.

File: test/e2e/specs/editor/various/fullscreen-mode.spec.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+		// Choose Full Screen Mode
+		await page
+			.locator( 'text=/Fullscreen modeWork without distraction/' )
+			.click();
```
Comment (Mamaduka) [inline test/e2e/specs/editor/various/fullscreen-mode.spec.js]:
```suggestion
		await page
			.locator( 'role=menuitemcheckbox', { hasText: 'Fullscreen mode' } )
			.click();
```

Nitpick: Let's use a role-based selector.

File: test/e2e/specs/editor/various/fullscreen-mode.spec.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+		// Validate Fullscreen
+		const isFullscreenEnabled = await page.$eval( 'body', ( body ) => {
+			return body.classList.contains( 'is-fullscreen-mode' );
+		} );
+
+		expect( isFullscreenEnabled ).toBe( true );
```
Comment (Mamaduka) [inline test/e2e/specs/editor/various/fullscreen-mode.spec.js]:
```suggestion
		// Check the body class.
		await expect( page.locator( 'body' ) ).toHaveClass(
			/is-fullscreen-mode/
		);
```

Suggestion: Let's use native API for class assertions - https://playwright.dev/docs/test-assertions#locator-assertions-to-have-class.

File: test/e2e/specs/editor/various/fullscreen-mode.spec.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+		const fullscreenCloseButton = await page.locator(
+			'.edit-post-fullscreen-mode-close'
+		);
+
+		expect( fullscreenCloseButton ).not.toBeNull();
```
Comment (Mamaduka) [inline test/e2e/specs/editor/various/fullscreen-mode.spec.js]:
```suggestion
		await expect(
			page.locator(
				'role=region[name="Editor top bar"i] >> role=link[name="View Posts"i]'
			)
		).toBeVisible();
```

It took me a bit to figure out the component with the `.edit-post-fullscreen-mode-close` class.

This part checks the "W" button in the top toolbar, so we can use a similar role-based selector. What do you think?

## Screenshot
CleanShot 2022-10-25 at 16 12 04

**Review by Mamaduka — COMMENTED** [review]
Thanks for contributing, @alvitazwar!

I left a few suggestions, which should help us make the e2e test more stable.

**Review by Mamaduka — COMMENTED** [review]
@alvitazwar, it looks like ESLint checks are failing. Let's fix that, and I think we can merge the PR.

**Review by Mamaduka — APPROVED** [review]
Thank you, @alvitazwar!

**Pull request merged**